src/ViewInstance.js

```diff
@@ -295,19 +328,7 @@ class ViewInstance extends React.Component {
           lastMenu={detailMenu}
           dismissible
           onClose={onClose}
-          actionMenuItems={[
-            {
-              id: 'edit-instance',
-              label: <FormattedMessage id="ui-inventory.editInstance" />,
-              href: this.craftLayerUrl('edit'),
-              onClick: this.onClickEditInstance,
-            },
-            {
-              id: 'copy-instance',
-              onClick: () => onCopy(instance),
-              label: <FormattedMessage id="ui-inventory.duplicateInstance" />,
-            }
-          ]}
+          actionMenu={(actionMenuProps) => this.getActionMenu(actionMenuProps, instance)}
```

Comment:
Here on every render cycle new function will be generated to be passed to `actionMenu` prop. It can be optimized with moving `(actionMenuProps) => this.getActionMenu(actionMenuProps, instance)` out of the render to a separate method.

Reply:
True. The reason why I decided to pass the `instance` constant was that it is generated above like so:

![image](https://user-images.githubusercontent.com/640976/48944824-90b03f00-ef28-11e8-845d-686e293b5a3b.png)

Without fully understanding the logic behind this I didn't want to duplicate this logic in the `getActionMenu`-method.

Would you recommend me doing so? Or maybe adding a method for this called `getInstance` that returns the `instance`?

Reply:
@cherewaty

Reply:
I like @Yurii-Danylenko's suggestion. That should work.

Reply:
Thanks – I'll make the change.